packages/mobile/src/paymentRequest/PaymentRequestConfirmation.tsx

```diff
@@ -17,92 +17,71 @@ import ContactCircle from 'src/components/ContactCircle'
 import CurrencyDisplay, { DisplayType } from 'src/components/CurrencyDisplay'
 import TotalLineItem from 'src/components/TotalLineItem'
 import { withTranslation } from 'src/i18n'
+import {
+  e164NumberToAddressSelector,
+  secureSendPhoneNumberMappingSelector,
+} from 'src/identity/selectors'
 import { emptyHeader } from 'src/navigator/Headers'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 import { writePaymentRequest } from 'src/paymentRequest/actions'
 import { PaymentRequestStatus } from 'src/paymentRequest/types'
 import { getDisplayName } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { ConfirmationInput, getConfirmationInput } from 'src/send/utils'
+import useSelector from 'src/redux/useSelector'
+import { getConfirmationInput } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import Logger from 'src/utils/Logger'
-import { currentAccountSelector } from 'src/web3/selectors'
+import { walletAddressSelector } from 'src/web3/selectors'
 
 // @ts-ignore
 const TAG = 'paymentRequest/confirmation'
 
-interface StateProps {
-  e164PhoneNumber: string | null
-  account: string | null
-  confirmationInput: ConfirmationInput
-  addressJustValidated?: boolean
-}
+type OwnProps = StackScreenProps<StackParamList, Screens.PaymentRequestConfirmation>
 
-interface DispatchProps {
-  showError: typeof showError
-  writePaymentRequest: typeof writePaymentRequest
-}
+type Props = WithTranslation & OwnProps
```

Comment:
I think the `WithTranslation` is a class-component thing, we can delete it and just use the `OwnProps` (maybe rename that one to `Props`)